src/netlink_namespace.c

```diff
@@ -0,0 +1,817 @@
+            "failed.");
+      return MNL_CB_ERROR;
+    }
+    ts = mnl_attr_get_payload(attr);
+
+    if (!stats_submitted && ts != NULL) {
+      char type_instance[DATA_MAX_NAME_LEN];
+
+      ssnprintf(type_instance, sizeof(type_instance), "%s-%s", tc_type,
+                tc_inst);
+
+      submit_one(dev, "ipt_bytes", type_instance, ts->bytes);
+      submit_one(dev, "ipt_packets", type_instance, ts->packets);
+    }
+
+    break;
+  }
+
+#endif /* TCA_STATS */
+
+#if !(HAVE_TCA_STATS && HAVE_TCA_STATS2)
+  DEBUG("netlink_namespace plugin: qos_filter_cb: Have neither TCA_STATS2 nor "
+        "TCA_STATS.");
+#endif
+
+  return MNL_CB_OK;
+} /* int qos_filter_cb */
+
+static int ir_config(const char *key, const char *value) {
+  char *new_val;
+  char *fields[8];
+  int fields_num;
+  int status = 1;
+  char nspath[1024] = "/var/run/netns/";
+
+  new_val = strdup(value);
+  if (new_val == NULL)
+    return (-1);
+
+  fields_num = strsplit(new_val, fields, STATIC_ARRAY_SIZE(fields));
+  if ((fields_num < 1) || (fields_num > 9)) {
+    sfree(new_val);
+    return (-1);
+  }
+
+  if ((strcasecmp(key, "Interface") == 0) ||
+      (strcasecmp(key, "VerboseInterface") == 0)) {
+    if (fields_num != 1) {
+      ERROR("netlink_namespace plugin: Invalid number of fields for option "
+            "`%s'. Got %i, expected 1.",
+            key, fields_num);
+      status = -1;
+    } else {
+      add_ignorelist(fields[0], "interface", NULL);
+      if (strcasecmp(key, "VerboseInterface") == 0)
+        add_ignorelist(fields[0], "if_detail", NULL);
+      status = 0;
+    }
+  } else if ((strcasecmp(key, "QDisc") == 0) ||
+             (strcasecmp(key, "Class") == 0) ||
+             (strcasecmp(key, "Filter") == 0)) {
+    if ((fields_num < 1) || (fields_num > 2)) {
+      ERROR("netlink_namespace plugin: Invalid number of fields for option "
+            "`%s'. Got %i, expected 1 or 2.",
+            key, fields_num);
+      return (-1);
+    } else {
+      add_ignorelist(fields[0], key, (fields_num == 2) ? fields[1] : NULL);
+      status = 0;
+    }
+  } else if (strcasecmp(key, "IgnoreSelected") == 0) {
+    if (fields_num != 1) {
+      ERROR("netlink_namespace plugin: Invalid number of fields for option "
+            "`IgnoreSelected'. Got %i, expected 1.",
+            fields_num);
+      status = -1;
+    } else {
+      if (IS_TRUE(fields[0]))
+        ir_ignorelist_invert = 0;
+      else
+        ir_ignorelist_invert = 1;
+      status = 0;
+    }
+  }
+  else if (strcasecmp(key, "NSName") == 0) {
+    INFO ("netlink_namespace plugin: recevied NSName value %s", fields[0]);
+    if ((fields_num < 1) || (fields_num > 2)) {
+      ERROR ("netlink_namespace plugin: Invalid number of fields for option "
+          "`%s'. Got %i, expected 1 or 2.", key, fields_num);
+      return (-1);
+    } else {
+      // ToDo: Check that /var/.... file exists (debuging messages please)
+      strncat(nspath, fields[0], sizeof(nspath) - strlen(nspath));
```

Comment:
You could reasonably demand that the user specifies the full path to the network namespace, as there's NOTHING that forces you to put your bound-to-network-namespace-files in /var/run/netns. While convention states it goes into /var/run/netns, I could also put it at /root/supersecretnetnsfile.